File: labs/Cipher-Encryption/Cipher.java
public class Cipher {
    // field
    private int key;

    // constructor
    public Cipher(int key) {
        this.key = key;
    }

    // encrypt method
    public String encrypt(String message) {
        StringBuilder encrypted = new StringBuilder();

        for (int i = 0; i < message.length(); i++) {
            char c = message.charAt(i);

            if (c >= 'a' && c <= 'z') {
                c += key;
                if (c > 'z') {
                    c -= 26;
                }
            } else if (c >= 'A' && c <= 'Z') {
                c += key;
                if (c > 'Z') {
                    c -= 26;
                }
            }
            encrypted.append(c);
        }
        return encrypted.toString();
    }

    // decrypt method
    public String decrypt(String encryptedMessage) {
        StringBuilder decrypted = new StringBuilder();

        for (int i = 0; i < encryptedMessage.length(); i++) {
            char c = encryptedMessage.charAt(i);

            if (c >= 'a' && c <= 'z') {
                c -= key;
                if (c < 'a') {
                    c += 26;
                }
            } else if (c >= 'A' && c <= 'Z') {
                c -= key;
                if (c < 'A') {
                    c += 26;
                }
            }
            decrypted.append(c);
        }
        return decrypted.toString();
    }
}

File: labs/Cipher-Encryption-CSharp/lab23/Cipher.cs
using System;
using System.Text;

public class Cipher
{
    private int _key;

    public int Key
    {
        get{ 
            return _key; 
            }
        set{
            _key = value;

            while (_key < 0)
                _key += 26;

            while (_key > 25)
                _key -= 26;
        }
    }

    // Encrypt method
    public string Encrypt(string message)
    {
        StringBuilder encrypted = new StringBuilder();

        for (int i = 0; i < message.Length; i++)
        {
            char c = message[i];

            if (c >= 'a' && c <= 'z')
            {
                c = (char)(c + _key);
                if (c > 'z')
                    c = (char)(c - 26);
            }
            else if (c >= 'A' && c <= 'Z')
            {
                c = (char)(c + _key);
                if (c > 'Z')
                    c = (char)(c - 26);
            }

            encrypted.Append(c);
        }

        return encrypted.ToString();
    }

    // Decrypt method
    public string Decrypt(string encryptedMessage)
    {
        StringBuilder decrypted = new StringBuilder();

        for (int i = 0; i < encryptedMessage.Length; i++)
        {
            char c = encryptedMessage[i];

            if (c >= 'a' && c <= 'z')
            {
                c = (char)(c - _key);
                if (c < 'a')
                    c = (char)(c + 26);
            }
            else if (c >= 'A' && c <= 'Z')
            {
                c = (char)(c - _key);
                if (c < 'A')
                    c = (char)(c + 26);
            }

            decrypted.Append(c);
        }

        return decrypted.ToString();
    }
}
